src/lib/realmdb/__tests__/UserProfileStorage.js

```diff
@@ -0,0 +1,455 @@
+    expect(value).toEqual(undefined)
+  })
+
+  it('should get display profile', () => {
+    const displayProfile = userProfileStorage.getDisplayProfile()
+    const { profile } = userProfileStorage
+    Object.keys(displayProfile).forEach(key => {
+      if (typeof displayProfile[key] !== 'function') {
+        expect(displayProfile[key]).toEqual(profile[key].display)
+      }
+    })
+  })
+
+  it('should get private profile', () => {
+    const privateProfile = userProfileStorage.getPrivateProfile()
+    const { profile } = userProfileStorage
+    Object.keys(privateProfile).forEach(key => {
+      if (typeof privateProfile[key] !== 'function') {
+        expect(privateProfile[key]).toEqual(profile[key].value)
+      }
+    })
+  })
+
+  it('should get fields privacy', () => {
+    Object.keys(userProfileStorage.profile).forEach(key => {
+      expect(userProfileStorage.profile[key].privacy).toEqual(userProfileStorage.getFieldPrivacy(key))
+    })
+  })
+
+  it('should pass profile validation', async () => {
+    const { isValid, errors } = await userProfileStorage.validateProfile(profile)
+    expect(isValid).toBeTruthy()
+    expect(errors).toEqual({})
+  })
+
+  it('should fail profile validation with empty profile', async () => {
+    const { isValid, errors } = await userProfileStorage.validateProfile()
+    expect(isValid).toBeFalsy()
+    expect(errors).toEqual({})
+  })
+
+  it('should fail profile validation with email error', async () => {
+    const { isValid, errors } = await userProfileStorage.validateProfile({
+      avatar: '',
+      email: '',
+      fullName: '',
+      mnemonic: '',
+      username: '',
+      mobile: '',
+      walletAddress: '',
+      smallAvatar: '',
+    })
+
+    // console.log(errors)
+    expect(isValid).toBeFalsy()
+    expect(errors).toEqual(expect.objectContaining({ email: 'Unavailable email' }))
+    expect(errors).toEqual(expect.objectContaining({ mnemonic: 'Unavailable mnemonic' }))
+    expect(errors).toEqual(expect.objectContaining({ username: 'Unavailable username' }))
+    expect(errors).toEqual(expect.objectContaining({ mobile: 'Unavailable mobile' }))
+    expect(errors).toEqual(expect.objectContaining({ walletAddress: 'Unavailable walletAddress' }))
+  })
+
+  it('should set profile field to private', async () => {
+    await userProfileStorage.setProfileFieldPrivacy('email', 'private')
+    expect(userProfileStorage.getFieldPrivacy('email')).toEqual('private')
+  })
+
+  it('should set profile field to masked', async () => {
+    await userProfileStorage.setProfileFieldPrivacy('email', 'masked')
+    expect(userProfileStorage.getFieldPrivacy('email')).toEqual('masked')
+  })
+
+  it('should set profile field to public', async () => {
+    await userProfileStorage.setProfileFieldPrivacy('email', 'public')
+    expect(userProfileStorage.getFieldPrivacy('email')).toEqual('public')
+  })
+
+  it('should find profile using getUserProfile with valid email', async () => {
```

Comment:
redudant tests
but also i'm not sure i understand what are they testing
getUserProfile isnt supposed to have any input